homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -65,18 +70,32 @@ async def async_step_user(self, user_input=None):
                 errors["base"] = "cannot_connect"
             except InvalidAuth:
                 errors["base"] = "invalid_auth"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(info["serial"])
+                await self.async_set_unique_id(info[KEY_SYS_SERIAL])
                 self._abort_if_unique_id_configured()
-                return self.async_create_entry(title=info["title"], data=user_input)
+                return self.async_create_entry(
+                    title=info[KEY_SYS_TITLE], data=user_input
+                )
 
         return self.async_show_form(
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=DATA_SCHEMA,
+            )
+        return await self.async_step_user()
```

Comment:
Shouldn't we pass the input to the user step? Otherwise we'll ask the user twice for the same thing.

Reply:
I cribbed the async_step_reauth_confirm flow from some other integrations. Please let me know if this is OK now.

(in https://github.com/home-assistant/core/pull/99411)